backend/src/test/java/botobo/core/domain/workbook/WorkbookSearchRepositoryTest.java

```diff
@@ -0,0 +1,320 @@
+package botobo.core.domain.workbook;
+
+import botobo.core.config.QuerydslConfig;
+import botobo.core.domain.card.Card;
+import botobo.core.domain.card.Cards;
+import botobo.core.domain.heart.Heart;
+import botobo.core.domain.tag.Tag;
+import botobo.core.domain.tag.Tags;
+import botobo.core.domain.user.Role;
+import botobo.core.domain.user.User;
+import botobo.core.domain.user.UserRepository;
+import botobo.core.ui.search.WorkbookSearchParameter;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.data.domain.Page;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@DataJpaTest(showSql = false)
+@ActiveProfiles("test")
+@Import({WorkbookSearchRepository.class, QuerydslConfig.class})
+class WorkbookSearchRepositoryTest {
+
+    @Autowired
+    private WorkbookRepository workbookRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private WorkbookSearchRepository workbookSearchRepository;
+
+    private Tag javaTag, javascriptTag;
+    private User bear, oz;
+
+    @BeforeEach
+    void setUp() {
+        initWorkbook(3);
+    }
+
+    @Test
+    @DisplayName("검색어를 입력하면 문제집 이름에서 검색어가 포함된 것들을 가져온다.")
+    void searchAllFromKeyword() {
+        // given
+        WorkbookSearchParameter parameter = WorkbookSearchParameter.builder()
+                .searchKeyword("Java")
+                .build();
+
+        // when
+        Page<Workbook> workbooks = workbookSearchRepository.searchAll(parameter, null, null, parameter.toPageRequest());
+        List<Workbook> workbookList = workbooks.toList();
+
+        // then
+        assertThat(workbookList).hasSize(6);
+        assertThat(workbookList.stream()
+                .map(Workbook::getName)
+                .allMatch(name -> name.contains("Java"))).isTrue();
```

Comment:
```suggestion
        assertThat(workbookList).extracting(Workbook::getName)
                .allMatch(name -> name.contains("Java"));
```
저번에 중간곰도 조앤 리뷰에 남겨줬는데 이렇게 extracting 을 활용해서 테스트를 더 깔끔하게 만들 수 있을 것 같아요! 아 그리고 isTrue 없이도 테스트가 잘 돌아가더라구요 😁

만약 extracting으로 바꾼다면 이 테스트 클래스 전체적으로 변경이 필요하네요.

Reply:
오오오!! 테스트에 대한 피드백이 필요했었는데 감삼다~~!! 전체적으로 extratcting을 사용하도록 수정했습니다!!